src/Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs

```diff
@@ -397,9 +397,12 @@ public void Stop()
 
                     if (blockValidationContext.Error == ConsensusErrors.BadTransactionDuplicate)
                     {
-                        int banDuration = blockValidationContext.BanDurationSeconds == BlockValidationContext.BanDurationDefaultBan ? this.connectionManager.ConnectionSettings.BanTimeSeconds : blockValidationContext.BanDurationSeconds;
-                        this.peerBanning.BanPeer(blockValidationContext.Peer, banDuration, $"Invalid block received: {blockValidationContext.Error.Message}");
                         this.logger.LogTrace("(-)[BAD_TX_DUP]");
+                        if (blockValidationContext.Peer == null) return;
```

Comment:
I think it will probably be a good idea to have a smaller check on line 419 in order to avoid duplicated code. 

So instead of that block you can do on line 419:
```
if (blockValidationContext.Error == ConsensusErrors.BadTransactionDuplicate)
{
 //log     
 return;
}
```

just before we ban the block.

Reply:
or just add 
`if (blockValidationContext.Error != ConsensusErrors.BadTransactionDuplicate)`
there and make no return (the method ends normally)

Reply:
OK Not sure what is the agreed solution reading that now : last time I discussed with @noescape00 we wanted to put that logic here to not have to do this bit
![image](https://user-images.githubusercontent.com/4638821/39425187-2f1ef172-4c72-11e8-83d2-7fab169c72d9.png)
So maybe to avoid code duplication I can simply methodise that logic and reuse the method twice ?